7_Classes/singleton_class.py

```diff
@@ -28,18 +28,32 @@ class SingletonMeta(type):
 
     _instances = {}
 
+    @staticmethod
+    def _hash_input(data):
+        import hashlib
+        import json
+
+        try:
+            dict_hash = hashlib.md5()
+            encoded = json.dumps(data[1], sort_keys=True).encode()
+            dict_hash.update(encoded)
+            return dict_hash.hexdigest()
+
+        except TypeError:
+            return data
+
     def __call__(cls, *args, **kwargs):
         """
         Possible changes to the value of the `__init__` argument do not affect
         the returned instance.
         """
 
         key = (cls, args[0])
```

Comment:
I know it's a test example, but could you please explain why you use `cls` here as a part of the key, but in the `_hash_input` you take the second element in the list `data[1]`?